koap/src/commonMain/kotlin/Message.kt

```diff
@@ -200,6 +201,7 @@ sealed class Message {
         /** RFC 7252 5.10.4. Accept */
         data class Accept(val format: Long) : Option() {
 
+            @JsName("constructorWithContentFormat")
```

Comment:
With this annotation, how is this accessed from JS?

_I feel like there is probably a better name, as you generally don't have "constructor" in function names._

Reply:
I was hoping to get some suggestions for the naming or a different strategy here TBH. The `JsName` here will map to a class function at the immediate parent scope, so in this case it would be:
```js
import koapModule from "..."
let acceptOption = koapModule.Message.Option.constructorWithContentFormat(0)
```

This is clearly terrible 🤪 

We should probably find a convention for these, perhaps `Message.Option.newAcceptOptionWithContentFormat`?

Reply:
Aw, I see the dilemma now.

Being that JS isn't type-safe we can't use type overloading. 😢 

If we export the primary constructor only (`data class Accept(val format: Long)`) will the secondary constructor (`constructor(format: ContentFormat)`) be hidden in JS?

I ask, because `ContentFormat` itself is just a wrapper around the `format` Long value, so if we can just hide the secondary constructor in JS then JS users could:

```typescript
import koapModule from "..."
let acceptOption = koapModule.Message.Option.Accept(0)

// or
let contentFormat = koapModule.Message.Option.ContentFormat(0)
koapModule.Message.Option.Accept(contentFormat.format)
```

---

If we can't only have the primary constructor visible, then my next idea would be to drop the secondary constructor (in common) then add an extension function in `jvmMain`:

```kotlin
fun Message.Option.Accept(format: ContentFormat) = Accept(format.format)
```

Reply:
> If we export the primary constructor only (data class Accept(val format: Long)) will the secondary constructor (constructor(format: ContentFormat)) be hidden in JS?

Both constructors are exported due to the enclosing class being annotated for export. It doesn't seem to be possible to hide a specific nested symbol (i.e. override the enclosing export).

The primary constructor is bound to the object initializer, so in javascript they would look like this:
```js
import koapModule from "..."
// primary (requires `new` keyword)
let acceptOption1 = new koapModule.Message.Option.Accept(0)
// secondary (class method)
let acceptOption2 = koapModule.Message.Option.newAcceptOptionWithContentFormat(
    new koapModule.Message.Option.ContentFormat(0)
)
```

> If we can't only have the primary constructor visible, then my next idea would be to drop the secondary constructor (in common) then add an extension function in `jvmMain`

Perhaps. But then we should consider if it is needed at all rather than introduce asymmetric APIs. If keeping it has merit, should we refactor things such that js consumers are provided with a nicer/safer interface in its place? ¯\\_(ツ)_/¯ 

Again, questions that lead to rethinking and refactoring which I'm deliberately avoiding in this PR. :smile: Answering coherently requires studying the use-cases as there are likely knock-on effects. The goal here is to enable the feature so that the compiler can insert its opinions during future adjustments to the code.

Reply:
Considering the goal of not doing a major refactor, I propose we give it a name like we would from a companion object. In this case, `fromContentFormat`.